src/v2/view-builder/views/EnrollProfileView.js

```diff
@@ -29,6 +29,16 @@ const Body = BaseForm.extend({
 
   },
   saveForm() {
+    const uiSchema = this.getUISchema();
+    _.each(this.model.attributes, (value, name) => {
+      if (_.isEmpty(value)) {
+        const uiSchemaProperty = uiSchema.find(schema => schema.name === name);
+        if (!_.isUndefined(uiSchemaProperty) && !uiSchemaProperty.required) {
```

Comment:
instead of deleting all required empty attributes, let's make it specific to which element you must delete if it is empty..

Also, we can make a generalized function in `AppState.js` and use it in this view... 

```
cleanModelState(attr) {
  // sanitize or delete here, attr = []
}
```

what to do you think ?

Reply:
Hello @pradeepdewda-okta, 

I like your idea about making a generalized function in `AppState.js`.

Regarding the first comment, I am deleting all non required attributes (optional attributes) from the model. At the beginning I thought this problem was exclusive for attributes rendered as dropdown, however after a lot of tests and checking other existing bugs I figure out that this problem is happening for attributes rendered as Text box too.

What I can do is to add an additional check so this deletion can happen only for dropdown and textbox.